### PR TITLE
Adding chatproxy_presence aka buddyList

### PR DESCRIPTION
**Pull request by GAMELASTER**
Returning a list of people (similar to presence) with type of status and timestamp. This event trigger on first pull and in some intervals (server-side, cannot be triggered manually)

BTW. both (chatproxy_presence and buddylist_overlay) returning a field "vc" (in chatproxy_presence).
As far I investigated it have something with status "Active on Messenger"

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -696,6 +696,15 @@ function getType(obj) {
   return Object.prototype.toString.call(obj).slice(8, -1);
 }
 
+function formatProxyPresence(presence, userID) {
+  if(presence.lat === undefined) return null;
+  return {
```
Comment (Schmavery) [inline utils.js]: Can we add a `type: "presence" to this to make it match the signature of the other event?

**Comment by Schmavery** [comment]
How is this different than buddylist_overlay?  Will we get duplicate notifications of presence now?

**Comment by GAMELASTER** [comment]
"buddylist_overlay" is event that sending the state changes for friends, for make texts like "Active 5 minutes ago" actual. "chatproxy_presence" is something like left-side bar with all friends and it's activity time. This contains (probably) all friends with their statuses and times :-) .

**Comment by GAMELASTER** [comment]
image
buddy_list overlay still contains only one user and their status update
image
chatproxy_presence with many many entries.

**Comment by GAMELASTER** [comment]
bump

**Comment by bsansouci** [comment]
Hey @GAMELASTER Thanks for doing this!
I think we could merge as soon as you address @Schmavery 's comment on the `utils.js` file :)

File: utils.js (hunk 1 above)
Comment (GAMELASTER) [inline utils.js]: As I written in my reply in pull request, the `type: presence` returns only ONE user presence, this return an list of all users and their presences. For make it match with the signature of default presence event, every user's presence in array will must call callback, what is not good idea.

**Comment by GAMELASTER** [comment]
@bsansouci 
Thanks for reply and sorry, my bad, I replied @Schmavery 's comment in [this](https://github.com/Schmavery/facebook-chat-api/pull/413#issuecomment-284273599) message in PR, but anyway I replied again on his comment.

File: utils.js [hunk 2]
```diff
@@ -696,6 +696,15 @@ function getType(obj) {
...
+    timestamp: presence.lat * 1000,
+    userID: userID,
+    statuses: presence.p === undefined ? 0 : presence.p
```
Comment (bsansouci) [inline utils.js]: Can you call this `status` instead.

**Comment by bsansouci** [comment]
Oh I see. Are you still able to get buddylist_overlay nowadays?
Btw could you add documentation that explains that `status: 0` is "not connected" and `status: 2` is connected (it seems from experimentation). I couldn't find any friends with `status: 1` so I don't know what that is lol.

**Comment by bsansouci** [comment]
Oh I just noticed it seems like they both contain timestamps. Do you think we could merge buddy list overlay and proxy-presence? We could transform both to be just `presence`?
If you make the `presence` flow return an array of items, they we could unify them right? So the buddylistoverlay will generally be an array containing just one item and this one will contain a bunch of items. But the information would be the same. What do you think?

**Comment by bsansouci** [comment]
Oh wait, I just noticed that we already document `0` and `2` statuses. So really I think we should unify the current presence and this new one :)

File: utils.js (hunk 2 above)
Comment (GAMELASTER) [inline utils.js]: I leaved it `statuses` because default `presence` object have it as `statuses`

**Comment by GAMELASTER** [comment]
@bsansouci we can unify them, but it will break scripts which working with actual format of presences (so it will broke back-compatilibity). So now I don't know what will be better

**Comment by GAMELASTER** [comment]
So well, how to solve the problem? Change the format of presence or keep back-compatilibity?

**Comment by Schmavery** [comment]
Sorry @GAMELASTER, I thought I responded to this but I guess I must not have.  I think right now it makes sense to iterate through the array and call the callback for each presence object.  If this becomes a performance problem, I think we can address that in the future.  Does that make sense?

**Comment by GAMELASTER** [comment]
@Schmavery I don't think it's performance issue, but it's just not looking good for me, but it will works and keeps back compatilibity. I'm going to change it a commit.

**Comment by GAMELASTER** [comment]
@Schmavery 
Tested, I think it's ready to merge 😉 😊

**Comment by Schmavery** [comment]
Thanks 😄

**Comment by wayfaredark** [comment]
I got a question and is how can i get the user fiend presence info ??? How exactly  is the url structure to perform request!!!!

**Comment by wayfaredark** [comment]
I perform request using https://0-edge-chat.facebook.com/pull, and asend headers to get frend user info but request callback say "INVALID REQUEST"

**Pull request merged**